applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_cr_beam_element_3D2N.cpp

```diff
@@ -28,6 +29,76 @@ AdjointFiniteDifferenceCrBeamElement::~AdjointFiniteDifferenceCrBeamElement()
 {
 }
 
+void AdjointFiniteDifferenceCrBeamElement::Calculate(const Variable<Vector >& rVariable, Vector& rOutput, const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // The particular soltuion of the influence function is rotated in global direction
+    if (rVariable == ADJOINT_PARTICULAR_DISPLACEMENT)
+    {
+        static constexpr int number_of_nodes = 2;
+        static constexpr int dimension = 3;
+        static constexpr unsigned int local_size = number_of_nodes * dimension;
+        static constexpr unsigned int element_size = local_size * 2;
+
+        KRATOS_ERROR_IF(rOutput.size() != element_size) << "Size of particular solution does not fit!" << std::endl;
+
+        CrBeamElement3D2N::Pointer p_primal_beam_element = dynamic_pointer_cast<CrBeamElement3D2N>(this->pGetPrimalElement());
```

Comment:
please do not do casts, refactor the existing functions instead (I recently started moving common functionalities to strucutral_mechancis_element_utilities)